node/src/proposal.ts

```diff
@@ -117,4 +123,29 @@ export class ProposalImpl implements Proposal {
         const signatureLength = this.#proposal.signature?.length ?? 0;
         return signatureLength > 0;
     }
+
+    private newEvaluateRequest(): gateway.IEvaluateRequest {
+        return {
+            transaction_id: this.#proposedTransaction.transaction_id,
+            channel_id: this.#channelName,
+            proposed_transaction: this.#proposal,
+        };
+    }
+
+    private newEndorseRequest(): gateway.IEndorseRequest {
+        return {
+            transaction_id: this.#proposedTransaction.transaction_id,
+            channel_id: this.#channelName,
+            proposed_transaction: this.#proposal,
+        };
+    }
+
+    // TODO Do something about "| null | undefined"???
+    private newPreparedTransaction(envelope: common.IEnvelope | null | undefined, result: protos.IResponse | null | undefined): gateway.IPreparedTransaction {
```

Comment:
These `| null | undefined` look a bit suspect to me- is there a better way to handle that?